__tests__/app.test.js

```diff
@@ -1,217 +1,301 @@
+          expect(comment).toHaveProperty("created_at", expect.any(String));
+          expect(comment).toHaveProperty("author", expect.any(String));
+          expect(comment).toHaveProperty("body", expect.any(String));
+          expect(comment).toHaveProperty("article_id", expect.any(Number));
+        });
+      });
+  });
+  test("status:200, should return an array of comments for the given article_id, ordered by date in descending order", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body;
+        expect(comments).toBeSortedBy("created_at", { descending: true });
+      });
+  });
+  test("status:404, should return an error msg with Not found when the ID does not exist", () => {
+    return request(app)
+      .get("/api/articles/999999/comments")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Not found");
+      });
+  });
+  test("status:400, should return an error msg with Bad request when the ID is not valid", () => {
+    return request(app)
+      .get("/api/articles/notAnId/comments")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
+});
 
-describe('GET /api/users', () => {
-    test('status:200, should return an array of user objects, with username, name and avatar_url properties', () => {
-        return request(app)
-        .get('/api/users')
-        .expect(200)
-        .then(({body}) => {
-            const { users } = body
-                expect(users).toHaveLength(4)
-            users.forEach((user) => {
-                expect(user).toHaveProperty('username', expect.any(String))
-                expect(user).toHaveProperty('name', expect.any(String))
-                expect(user).toHaveProperty('avatar_url', expect.any(String))
-            })
-        })
-    })
-    test('status:404, should return an error msg Not found when passed a route that does not exist', () => {
+describe("POST /api/articles/:article_id/comments", () => {
+  test("status:201, should add a comment for an article when passed an article ID. Should pass the correct comment ID when passed a subsequent comment.", () => {
+    const newComment1 = {
+      body: "An insightful article shedding light on the latest breakthrough in renewable energy technology!",
+      username: "butter_bridge",
+    };
+    return request(app)
+      .post("/api/articles/4/comments")
+      .send(newComment1)
+      .expect(201)
+      .then(({ body }) => {
+        const { comment } = body;
+        expect(comment).toMatchObject({
+          comment_id: 19,
+          body: "An insightful article shedding light on the latest breakthrough in renewable energy technology!",
+          article_id: 4,
+          author: "butter_bridge",
+          votes: 0,
+          created_at: expect.any(String),
+        });
+        const newComment2 = {
```

Comment:
Technically we don't need to worry about duplicating the same behaviour and testing for it again in our tests